thrust/thrust/system/cuda/detail/core/triple_chevron_launch.h

```diff
@@ -71,38 +72,45 @@ struct _CCCL_VISIBILITY_HIDDEN triple_chevron
   template <class T>
   size_t _CCCL_DEVICE align_up(size_t offset) const
   {
-    size_t alignment = alignment_of<T>::value;
-    return alignment * ((offset + (alignment - 1)) / alignment);
+    return ::cuda::ceil_div(offset, alignof(T)) * alignof(T);
   }
 
   size_t _CCCL_DEVICE argument_pack_size(size_t size) const
   {
     return size;
   }
-  template <class Arg, class... Args>
-  size_t _CCCL_DEVICE argument_pack_size(size_t size, Arg const& arg, Args const&... args) const
+
+  template <class... Args>
+  size_t _CCCL_DEVICE argument_pack_size(size_t size, Args const&...) const
   {
-    size = align_up<Arg>(size);
-    return argument_pack_size(size + sizeof(Arg), args...);
+    // TODO(bgruber): replace by fold over comma in C++17 (make sure order of evaluation is left to right!)
+    int dummy[] = {(size += align_up<Args>(size) + sizeof(Args), 0)...};
+    (void) dummy;
+    return size;
   }
 
   template <class Arg>
-  size_t _CCCL_DEVICE copy_arg(char* buffer, size_t offset, Arg arg) const
+  void _CCCL_DEVICE copy_arg(char*& buffer, size_t& offset, Arg arg) const
```

Comment:
Why do we need to pass `buffer` by reference here?

Reply:
Good catch! I don't. I previously tried to move the pointer instead, but concluded to go with moving the `offset`.